src/tactic/squeeze.lean

```diff
@@ -30,10 +30,9 @@ namespace tactic
 -/
 meta def erase_simp_args (hs : list simp_arg_type) (s : name_set) : tactic name_set :=
 do
-  -- TODO: when Lean 3.4 support is dropped, use `decode_simp_arg_list_with_symm` on the next line:
-  (hs, _, _) ← decode_simp_arg_list hs,
-  pure $ hs.foldr (λ (h : pexpr) (s : name_set),
-    match h.get_app_fn h with
+  (hs, _, _) ← decode_simp_arg_list_with_symm hs,
+  pure $ hs.foldr (λ (h : pexpr × bool) (s : name_set),
+    match h.1.get_app_fn h.1 with
```

Comment:
Can you explain why this match works?  At first glance it looks like `h.1.get_app_fn h.1` elaborates to `⇑h.1.get_app_fn h.1` and hence never matches `expr.const _ _` because it is an `expr.app`.

Reply:
I believe you are correct that the match doesn't work, but I'm unsure what the exact intention of the function is (the match was like this when this code was introduced in PR #1047). It should remove duplicates when a `@[simp]` lemma is explicitly added to the arguments, right? But I haven't been able to construct a test case for `squeeze_simp` where I saw a difference between `h.1.get_app h.1` and with `h.1.get_app` - both versions remove duplicate/unneeded lemmas.

Reply:
Can we just erase the `erase_simp_args` function then?

Reply:
Looks like #2218 changed `squeeze_simp` so that any duplicate/unneeded lemmas will always get deleted by the call to `filter_simp_set` after that. My conclusion: erase `erase_simp_args`.